lib/execution_engine2/sdk/EE2Runjob.py

```diff
@@ -460,6 +504,130 @@ def _check_job_arguments(self, jobs, has_parent_job=False):
+            terminated_code=TerminatedCode.terminated_by_user_retry.value,
+            as_admin=as_admin,
+        )
+
+        # Submit job to job scheduler or fail and not count it as a retry attempt
+        run_job_params[_PARENT_RETRY_JOB_ID] = job_id
+        child_job_id = self.run(params=run_job_params, as_admin=as_admin)
+
+        # Save that the job has been retried, and increment the count
+        # TODO Use and create a method in sdkmr?
+        job.modify(inc__retry_count=1, set__retried=True)
+
+        # Should we compare the original and child job to make sure certain fields match,
+        # to make sure the retried job is correctly submitted? Or save that for a unit test?
+        return child_job_id
+
+    @staticmethod
+    def _get_job_input_params_from_existing_job(job_input: JobInput) -> Dict:
+        """
+        Get input level fields.
+        """
+        # Expected fields are found in `_init_job_rec`
+        inputs = dict()
+
+        # Used to iterate over "Job" attributes but in dictionary like lookup
+        inputs_list = [
+            _METHOD,
+            _APP_PARAMS,
+            _SERVICE_VER,
+            _APP_ID,
+            _PARENT_JOB_ID,
+            _WORKSPACE_ID,
+            "requirements",
+        ]
+
+        # Insert field if available
+        for field in inputs_list:
+            if job_input[field]:
+                inputs[field] = job_input[field]
+
+        # _SOURCE_WS_OBJECTS list
+        if job_input[_SOURCE_WS_OBJECTS]:
+            inputs[_SOURCE_WS_OBJECTS] = list(job_input[_SOURCE_WS_OBJECTS])
+
+        # Special Cases: It is OK if meta/narr_cell_info is in params 2x
+        if job_input["narrative_cell_info"]:
+            # For easier testing
+            nci = job_input["narrative_cell_info"]  # type: Meta
+            inputs["meta"] = inputs["narrative_cell_info"] = nci.to_mongo().to_dict()
+
+        # Require a non blank params for job browser or narrative compatibility
+        if _APP_PARAMS not in inputs:
+            inputs[_APP_PARAMS] = {}
+
+        return inputs
+
+    @staticmethod
+    def _get_run_job_params_from_existing_job(job: Job, user_id: str) -> Dict:
+        """
+        Get top level fields from job model to be sent into `run_job`
+        """
+        ji = job.job_input  # type: JobInput
+        job_input = EE2RunJob._get_job_input_params_from_existing_job(ji)
+
+        run_job_params = {
+            _WORKSPACE_ID: job.wsid,  # optional
+            "meta": job_input.get(_NARRATIVE_CELL_INFO),  # optional
+            _APP_PARAMS: job_input.get(_APP_PARAMS),  # optional
+            "user": user_id,  # required, it runs as the current user
+            _METHOD: job_input[_METHOD],  # required
+            _APP_ID: job_input.get(_APP_ID),  # optional
+            # _JOB_INPUT: job_input,  # required
+            _SOURCE_WS_OBJECTS: job_input.get(_SOURCE_WS_OBJECTS),  # optional
+            _SERVICE_VER: job_input.get(_SERVICE_VER),  # optional
```

Comment:
is there a nicer way to do this so you're not repeating
```py
    _PARAM_NAME: job_input.get(_PARAM_NAME),
```
so many times?

Reply:
I had it in a dictionary lookup, but then some keys didn't match up, so I deleted the dictionary, ended up less lines